doc/developer-notes.md

```diff
@@ -88,15 +88,22 @@ required when doing so would need changes to significant pieces of existing
 code.
   - Variable (including function arguments) and namespace names are all lowercase and may use `_` to
     separate words (snake_case).
-    - Class member variables have a `m_` prefix.
-    - Global variables have a `g_` prefix.
+    - Class member variables have a `m_` prefix, so it's obvious when variable
+      references in the code implicitly access the current class instance.
+    - Global variables have a `g_` prefix, so it's obvious when code is
+      accessing global state.
   - Constant names are all uppercase, and use `_` to separate words.
   - Enumerator constants may be `snake_case`, `PascalCase` or `ALL_CAPS`.
     This is a more tolerant policy than the [C++ Core
     Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-caps),
     which recommend using `snake_case`.  Please use what seems appropriate.
-  - Class names, function names, and method names are UpperCamelCase
-    (PascalCase). Do not prefix class names with `C`.
+  - Type/class names, standalone function names, and static method names are
+    UpperCamelCase. Instance method names are encouraged to be lowerCamelCase, so 
+    it's obvious when function calls in the code implicitly access the current class
```

Comment:
Might be better to be shorter "so it's obvious when code is calling implicit instance member functions"